src/test/shared/sam/cli/samCliBuild.test.ts

```diff
@@ -0,0 +1,292 @@
+    })
+
+    it('Passes build command to sam cli', async () => {
+        const nonRelevantArg = 'arg is not of interest to this test'
+
+        const processInvoker: SamCliProcessInvoker = new TextProcessInvoker((args: any[]) => {
+            assert.ok(args.length > 0, 'Expected args to be present')
+            assert.strictEqual(args[0], 'build', 'Expected first arg to be the build command')
+        })
+
+        await new SamCliBuildInvocation({
+            buildDir: nonRelevantArg,
+            baseDir: nonRelevantArg,
+            templatePath: placeholderTemplateFile,
+            invoker: processInvoker,
+        }).execute()
+    })
+
+    it('Passes Build Dir to sam cli', async () => {
+        const expectedBuildDir = '/build'
+        const nonRelevantArg = 'arg is not of interest to this test'
+
+        const processInvoker: SamCliProcessInvoker = new TextProcessInvoker((args: any[]) => {
+            assertArgsContainArgument(args, '--build-dir', expectedBuildDir)
+        })
+
+        await new SamCliBuildInvocation({
+            buildDir: expectedBuildDir,
+            baseDir: nonRelevantArg,
+            templatePath: placeholderTemplateFile,
+            invoker: processInvoker,
+        }).execute()
+    })
+
+    it('Passes Base Dir to sam cli', async () => {
+        const expectedBaseDir = '/src'
+        const nonRelevantArg = 'arg is not of interest to this test'
+
+        const processInvoker: SamCliProcessInvoker = new TextProcessInvoker((args: any[]) => {
+            assertArgsContainArgument(args, '--base-dir', expectedBaseDir)
+        })
+
+        await new SamCliBuildInvocation({
+            buildDir: nonRelevantArg,
+            baseDir: expectedBaseDir,
+            templatePath: placeholderTemplateFile,
+            invoker: processInvoker,
+        }).execute()
+    })
+
+    it('Does not pass Base Dir to sam cli', async () => {
+        const nonRelevantArg = 'arg is not of interest to this test'
+
+        const processInvoker: SamCliProcessInvoker = new TextProcessInvoker((args: any[]) => {
+            assertArgNotPresent(args, '--base-dir')
+        })
+
+        await new SamCliBuildInvocation({
+            buildDir: nonRelevantArg,
+            templatePath: placeholderTemplateFile,
+            invoker: processInvoker,
+        }).execute()
+    })
+
+    it('Passes Template to sam cli', async () => {
+        const nonRelevantArg = 'arg is not of interest to this test'
+
+        const processInvoker: SamCliProcessInvoker = new TextProcessInvoker((args: any[]) => {
+            assertArgsContainArgument(args, '--template', placeholderTemplateFile)
+        })
+
+        await new SamCliBuildInvocation({
+            buildDir: nonRelevantArg,
+            baseDir: nonRelevantArg,
+            templatePath: placeholderTemplateFile,
+            invoker: processInvoker,
+        }).execute()
+    })
+
+    it('useContainer = true passes --use-container to sam cli', async () => {
```

Comment:
Optional: I think this would be a more clear description: `passes --use-container to sam cli if useContainer is true`. Same throughout this file.